rts/System/SpringMath.cpp

```diff
@@ -109,6 +109,69 @@ bool ClosestPointOnRay(const float3 p0, const float3 ray, const float3 p, float3
 		return false;
 
 	px = p0 + ray * pdist;
+
+	return true;
+}
+
+
+// Credit:
+// - https://stackoverflow.com/a/38437831/7351594
+// - Practical Geometry Algorithms - Daniel Sunday
+// We use the 'Direct Linear Equation' method described in the book above
+float3 SolveIntersectingPoint(int zeroCoord, int coord1, int coord2, float4 p1, float4 p2)
+{
+	const float a1 = p1[coord1];
+	const float b1 = p1[coord2];
+	const float d1 = -p1[3];
+
+	const float a2 = p2[coord1];
+	const float b2 = p2[coord2];
+	const float d2 = -p2[3];
+
+	float3 point;
+
+	point[zeroCoord] = 0;
+	point[coord1] = (b2 * d1 - b1 * d2) / (a1 * b2 - a2 * b1);
+	point[coord2] = (a1 * d2 - a2 * d1) / (a1 * b2 - a2 * b1);
+
+	return point;
+}
+
+
+// This method helps finding a point on the intersection between two planes.
+// Depending on the orientation of the planes, the problem could solve for the
+// zero point on either the x, y or z axis
+bool IntersectPlanes(const float4& p1, const float4& p2, std::pair<float3, float3> &line)
+{
+	// the cross product gives us the direction of the line at the intersection
+	// of the two planes, and gives us an easy way to check if the two planes
+	// are parallel - the cross product will have zero magnitude
+	const float3 normal1 = p1;
+	const float3 normal2 = p2;
+
+	const float3 direction = normal1.cross(normal2).UnsafeANormalize();
+	const float magnitude = direction.Length();
```

Comment:
Would be great if we could skip the square root here, I just ported it from the original impl. Perhaps if we are assured normal1 and normal2 are normalized we can just dot them and check if close to `1.0f`